src/update-feeds/update-feeds.ts

```diff
@@ -124,8 +138,82 @@ export const runUpdateFeed = async (providerName: string, chain: Chain, chainId:
+        item.dataFeed.dataFeeds.map((dataFeed) => ({
+          url: `${url}/${dataFeed.airnodeAddress}`,
+          lastReceived: Date.now(),
+        }))
+      );
+
+      const freshExistingUrls = draft.signedApiUrlStore.filter(
+        (url) => Date.now() - url.lastReceived > 1000 * 60 * SIGNED_URL_EXPIRY_IN_MINUTES
+      );
+
+      // Appends records that don't already exist, updates records that already exist with new timestamps
+      draft.signedApiUrlStore = mergeWith(receivedUrls, freshExistingUrls, (a, b) => (a.url === b.url ? a : b));
+
+      const cachedDapiResponse = draft.dapis[item.dapiName];
+      // We're assuming the received data feed value is newer than what we already have... this may not actually be the case
+      // but the alternative is that we accept a later value but then DoS future values.
+      // This should probably be time-constrained and require updated values (eg. only update if newer but not newer than 15 minutes)
+      draft.dapis[item.dapiName] = {
+        dataFeed: cachedDapiResponse?.dataFeed ?? item.dataFeed,
+        dataFeedValues: { ...cachedDapiResponse?.dataFeedValues, [item.chainId]: item.dataFeedValue },
+        updateParameters: { ...cachedDapiResponse?.updateParameters, [item.chainId]: item.updateParameters },
+      };
+    })
+  );
+};
+
+export const getFeedsToUpdate = (batch: ReadDapiWithIndexResponsesAndChainId) =>
+  batch.map((dapiResponse) => {
+    const signedData = getStoreDataPoint(dapiResponse.dataFeed.dataFeedId);
+
+    if (signedData === undefined) {
+      return { ...dapiResponse, shouldUpdate: false };
+    }
+
+    const offChainValue = BigNumber.from(signedData.encodedValue);
+    const offChainTimestamp = Number.parseInt(signedData?.timestamp ?? '0', 10);
+    const deviationThreshold = dapiResponse.updateParameters.deviationThresholdInPercentage;
+
+    const shouldUpdate = checkUpdateConditions(
+      dapiResponse.dataFeedValue.value,
+      dapiResponse.dataFeedValue.timestamp,
+      offChainValue,
+      offChainTimestamp,
+      dapiResponse.updateParameters.heartbeatInterval,
+      deviationThreshold
+    );
+
+    return {
+      ...dapiResponse,
+      shouldUpdate,
+      signedData,
+    };
+  });
+
+export const updateFeeds = async (_batch: ReturnType<typeof getFeedsToUpdate>) => {
+  // TODO implement
+  // batch, execute
+};
+
+export const processBatch = async (batch: ReadDapiWithIndexResponsesAndChainId) => {
   logger.debug('Processing batch of active dAPIs', { batch });
-  // TODO: Implement.
+
+  // Start by merging the dynamic state with the state
+  updateDynamicState(batch);
+
+  const allFeeds = getFeedsToUpdate(batch);
+
+  // Flow chart calls for clearing stale data
+  // "for each dAPI where an update is not needed" ... "clear stored on chain datafeed value from gas price store"
+  // TODO The fn called has been removed, what replaces it (if any)? - it may require sponsor wallet derivation
```

Comment:
I handled this [here](https://github.com/api3dao/airseeker-v2/pull/68) 😄

Reply:
Awesome, I'll remove this comment 👍

Reply:
I think the way batches are processed now is ok, we still need the ones not being updated as they're now 😄